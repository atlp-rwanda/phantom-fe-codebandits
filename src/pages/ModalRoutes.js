import React from 'react';
import { Route, Routes } from 'react-router-dom';
import RouteView from './routes/RouteView.js';

function ModalRoutes() {
  return (
    <div className="fixed top-0 left-0 right-0 bottom-0 bg-gray-400 bg-opacity-40 flex justify-center items-center overflow-hidden">
      <div className="h-fit w-fit overflow-auto rounded-md shadow-main">
        <Routes>
          <Route path="routes/view/:id" element={<RouteView />}></Route>
        </Routes>
      </div>
    </div>
  );
}

export default ModalRoutes;
