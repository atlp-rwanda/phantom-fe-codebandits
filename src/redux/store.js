import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer
  }
});
