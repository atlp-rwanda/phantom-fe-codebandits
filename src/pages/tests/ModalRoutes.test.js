import React from 'react';
import renderer from 'react-test-renderer';
import {  mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import ModalRoutes from '../ModalRoutes.js'
import { Provider } from 'react-redux';
import {store} from '../../redux/store.js'

describe('Modal Routes', () => {
  it('should render the ModalRoutes component', () => {
    const elem = renderer
      .create(
		  <Provider store={store}>
          <MemoryRouter>
            <ModalRoutes />
          </MemoryRouter>
		  </Provider>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should not render the Routes component', () => {
   const wrapper = mount(
	<Provider store={store}>
	<MemoryRouter>
	  <ModalRoutes />
	</MemoryRouter>
	</Provider>
   )
   expect(wrapper.find('Routes').length).toEqual(1)
  });

  it('should not render the Route component', () => {
	const wrapper = mount(
	 <Provider store={store}>
	 <MemoryRouter>
	   <ModalRoutes />
	 </MemoryRouter>
	 </Provider>
	)
	expect(wrapper.find('Route').length).toEqual(0)
   });
  

});
