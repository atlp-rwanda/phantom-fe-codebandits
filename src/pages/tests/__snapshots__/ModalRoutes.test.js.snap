// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Routes should render the ModalRoutes component 1`] = `
<div
  className="fixed top-0 left-0 right-0 bottom-0 bg-gray-400 bg-opacity-40 flex justify-center items-center overflow-hidden"
>
  <div
    className="h-fit w-fit overflow-auto rounded-md shadow-main"
  />
</div>
`;
